sources/packages/api/src/cra-integration/cra-files/cra-file-header.ts

```diff
@@ -30,4 +31,14 @@ export class CRAFileHeader implements CRAFileLine {
     header.append("0");
     return header.toString();
   }
+
+  public static CreateFromLine(line: string): CRAFileHeader {
+    const header = new CRAFileHeader();
+    header.transactionCode = line.substr(0, 4) as TransactionCodes;
+    header.processDate = dayjs(line.substr(28, 8), DATE_FORMAT).toDate();
```

Comment:
what is in any scenario line.substr(28, 8) is a null or empty string, will dayjs raise an error?

Reply:
The lines are going to be a fixed 147 long string. The date is part of the CRA response and is not optional, it will be in the response unless something really bad happens from CRA side.
If it happens, the date will be invalid, as shown below, but it will not throw an exception. Right now, it will not cause an issue in the process also.

![image](https://user-images.githubusercontent.com/61259237/120362276-c1268000-c2bf-11eb-8ca9-21c0506ef32a.png)